### PR TITLE
Increase default max alt names from 128 to 1024

### LINKED ISSUE #7756: [Bug]: The value of WOLFSSL_MAX_ALT_NAMES causes an error of ASN_ALT_NAME_E
### Contact Details

_No response_

### Version

5.7.2

### Description

I updated to version 5.7.2 via vcpkg but I got ASN_ALT_NAME_E error while handshaking YouTube domain (www.youtube.com)
The error is generated at this line:
https://github.com/wolfSSL/wolfssl/blob/ec9fcf53539755ed3a38e386a00d91478ff357b3/wolfcrypt/src/asn.c#L18986
due to numNames reaching 129 which is larger than WOLFSSL_MAX_ALT_NAMES
At this point idx = 2358 and sz = 2460 so there is still 102 bytes to decode

I don't know if numNames reaching 129 is normal and the bug is from the new introduced WOLFSSL_MAX_ALT_NAMES (as per the release notes!) or there is something wrong with the received certificate? The same code is working with previous versions

I tested handshake with different domains, and it is just working.

### PR DESCRIPTION
**Pull request by ColtonWilley**
# Description

Increase default max alt names from 128 to 1024
Fixes #7756

### CODE REVIEW

File: tests/api.c [hunk 1]
```diff
@@ -51497,7 +51497,7 @@ static int test_wolfSSL_X509_max_altnames(void)
 #if !defined(NO_FILESYSTEM) && !defined(NO_CERTS) && !defined(NO_RSA)
 
     /* Only test if max alt names has not been modified */
-#if WOLFSSL_MAX_ALT_NAMES == 128
+#if WOLFSSL_MAX_ALT_NAMES == 1024
```
Comment (dgarske) [inline tests/api.c:51500]: Could this be `#if WOLFSSL_MAX_ALT_NAMES <= 1024`?

File: wolfssl/wolfcrypt/asn.h [hunk 2]
```diff
@@ -799,7 +799,7 @@ extern const WOLFSSL_ObjectInfo wolfssl_object_info[];
  * Any certificate containing more than this number of subject
  * alternative names will cause an error when attempting to parse. */
 #ifndef WOLFSSL_MAX_ALT_NAMES
-#define WOLFSSL_MAX_ALT_NAMES 128
+#define WOLFSSL_MAX_ALT_NAMES 1024
```
Comment (dgarske) [inline wolfssl/wolfcrypt/asn.h:802]: Didn't we decide 512? I am okay with 1024 if you feel strongly about it

**Review by dgarske — CHANGES_REQUESTED** [review]

File: wolfssl/wolfcrypt/asn.h (hunk 2 above)
Comment (ColtonWilley) [reply wolfssl/wolfcrypt/asn.h:802]: I didnt remember if we decided on 512 or 1024 at the end. I would just hate to make this change and then hit another cert in the wild with 513 names only to have to change it again. 512 and 1024 are functionally the same in terms of denial of service protection to me, so I see no reason to limit at 512.

File: tests/api.c (hunk 1 above)
Comment (ColtonWilley) [reply tests/api.c:51500]: I guess I didnt think about somebody lowering the limit for some reason, yes that should definitely work I will make the change.

**Review by dgarske — APPROVED** [review]

**Comment by embhorn** [comment]
Fixes #7756

**Pull request merged**